task_pool.go

```diff
@@ -0,0 +1,141 @@
+package getty
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+)
+
+import (
+	log "github.com/AlexStocks/log4go"
+)
+
+//func init() {
+//	rand.Seed(time.Now().UnixNano())
+//}
+//
+//var (
+//	// The golang rand generators are *not* intrinsically thread-safe.
+//	randIDLock sync.Mutex
+//	randIDGen  = rand.New(rand.NewSource(time.Now().UnixNano()))
+//)
+//
+//func randID() uint64 {
+//	randIDLock.Lock()
+//	defer randIDLock.Unlock()
+//
+//	return uint64(randIDGen.Int63())
+//}
+
+// task t
+type task struct {
+	session *session
+	pkg     interface{}
+}
+
+// task pool: manage task ts
+type taskPool struct {
+	idx    int32
+	qLen   int32 // task t queue length
+	size   int32 // pool size
+	qArray []chan task
+	wg     sync.WaitGroup
+
+	once sync.Once
+	done chan struct{}
+}
+
+// build a task pool
+func newTaskPool(poolSize int32, taskQLen int32) *taskPool {
+	return &taskPool{
+		size:   poolSize,
+		qLen:   taskQLen,
+		qArray: make([]chan task, poolSize),
```

Comment:
- do u agree a buffer chan is a pool?  if not, why ?
- does it need to create a task pool array for every session?
- does it need to create a task pool for every session? why not creating a global pool?

Reply:
Firstly, the buffer chan is a array. It stores all task queue.
Secondly, it is wrong to create task pool array for every session. It should be created for every client/server.